WcaOnRails/app/helpers/competitions_helper.rb

```diff
@@ -112,4 +112,21 @@ def championship_option_tags(selected: nil)
     }
     grouped_options_for_select(grouped_championship_types, selected)
   end
+
+  def first_and_last_time_from_activities(activities, timezone)
+    sorted_activities = activities.sort_by { |a| a.start_time.in_time_zone(timezone).strftime("%H:%M") }
```

Comment:
It took me a few minutes to find out why I definitely needed to do it like this.
I'll add the following comment since it's definitely not trivial at first sight:
```ruby
    # The goal of this function is to determine what should be the starting and ending points in the time axis of the calendar.
    # Which means we need to find the earliest start_time (and latest end_time) for any activity occuring on all days, expressed in the local timezone.
    # To do that we first convert the start_time to the local timezone, and keep only the "time of the day" component of the datetime.
    # We can sort the activities based on this value to compute the extremum of the time axis.
```